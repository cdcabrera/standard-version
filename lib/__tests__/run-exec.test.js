const runExec = require('../run-exec');

describe('run-exec', () => {
  it('should attempt to exec commands', async () => {
    const mockWarn = jest.fn();
    const spy = jest.spyOn(console, 'warn').mockImplementation(mockWarn);

    const dryRunResults = await runExec({ dryRun: true });
    const echoCmdResults = await runExec({}, 'echo "lorem ipsum"');
    const errorResults = await runExec({}, 'echo "dolor sit error" && exit 0');
    const errorResultsStdOut = await runExec({}, 'echo "dolor sit stdout error" 1>&2 && exit 0');

    expect({
      dryRunResults,
      echoCmdResults,
      errorResults,
      errorResultsStdOut
    }).toMatchSnapshot('command results');

    expect(mockWarn.mock.calls).toMatchSnapshot('console messages');

    spy.mockClear();
  });
});
