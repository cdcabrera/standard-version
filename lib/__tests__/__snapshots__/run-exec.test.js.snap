// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`run-exec should attempt to exec commands: command results 1`] = `
Object {
  "dryRunResults": undefined,
  "echoCmdResults": "lorem ipsum
",
  "errorResults": "dolor sit error
",
  "errorResultsStdOut": "",
}
`;

exports[`run-exec should attempt to exec commands: console messages 1`] = `
Array [
  Array [
    "<yellow>dolor sit stdout error
</yellow>",
  ],
]
`;
